http/src/v0/block.rs

```diff
@@ -98,31 +102,174 @@ pub fn put<T: IpfsTypes>(
         .and_then(put_query)
 }
 
-#[derive(Debug, Deserialize)]
-pub struct RmQuery {
-    arg: String,
+#[derive(Debug)]
+pub struct RmOptions {
+    // Bash58 encoded multihashes of block(s) to remove
+    args: Vec<String>,
+    // Ignore nonexistent blocks
+    force: bool,
+    // Write minimal output
+    quiet: bool,
 }
 
 #[derive(Debug, Serialize)]
 #[serde(rename_all = "PascalCase")]
-pub struct RmResponse {}
-
-async fn rm_query<T: IpfsTypes>(ipfs: Ipfs<T>, query: RmQuery) -> Result<impl Reply, Rejection> {
-    let cid: Cid = query.arg.parse().map_err(StringError::from)?;
-    ipfs.remove_block(&cid).await.map_err(StringError::from)?;
-    let response = RmResponse {};
-    Ok(reply::json(&response))
+pub struct RmResponse {
+    error: String,
+    hash: String,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct EmptyResponse {}
+
 pub fn rm<T: IpfsTypes>(
     ipfs: &Ipfs<T>,
 ) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
     path!("block" / "rm")
         .and(with_ipfs(ipfs))
-        .and(query::<RmQuery>())
+        .and(rm_options())
         .and_then(rm_query)
 }
 
+use futures::stream::Stream;
+use std::borrow::Cow;
+use std::convert::TryFrom;
+use std::fmt;
+
+fn rm_options() -> impl Filter<Extract = (RmOptions,), Error = Rejection> + Clone {
+    warp::filters::query::raw().and_then(|q: String| {
+        let res = RmOptions::try_from(q.as_str())
+            .map_err(StringError::from)
+            .map_err(warp::reject::custom);
+
+        futures::future::ready(res)
+    })
+}
+
+#[derive(Debug)]
+pub enum ParseError<'a> {
+    DuplicateField(Cow<'a, str>),
+    MissingArg,
+    InvalidNumber(Cow<'a, str>, Cow<'a, str>),
+    InvalidBoolean(Cow<'a, str>, Cow<'a, str>),
+}
+
+impl<'a> fmt::Display for ParseError<'a> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        use ParseError::*;
+        match *self {
+            DuplicateField(ref s) => write!(fmt, "field {:?} was duplicated", *s),
+            MissingArg => write!(fmt, "required field \"arg\" missing"),
+            InvalidNumber(ref k, ref v) => write!(fmt, "field {:?} invalid number: {:?}", *k, *v),
+            InvalidBoolean(ref k, ref v) => write!(fmt, "field {:?} invalid boolean: {:?}", *k, *v),
+        }
+    }
+}
+
+impl<'a> TryFrom<&'a str> for RmOptions {
```

Comment:
could do similar splitting as I did with `refs`, move this to `block/options.rs`